compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -127,16 +129,14 @@ impl Inliner<'tcx> {
                 self.tcx.instance_mir(callsite.callee.def)
             };
 
-            let callee_body: &Body<'tcx> = &*callee_body;
+            if !self.consider_optimizing(callsite, &callee_body) {
+                continue;
+            }
 
-            let callee_body = if self.consider_optimizing(callsite, callee_body) {
-                self.tcx.subst_and_normalize_erasing_regions(
-                    &callsite.callee.substs,
-                    self.param_env,
-                    callee_body,
-                )
+            let callee_body = if let Some(substs) = callsite.callee.substs_for_mir_body() {
```

Comment:
There are like 5 duplications of this code spread around the various backends, the mir interpreter and the collector. Maybe create a method on `Instance` that takes a param env, a tcx and a generic arg that should be substituted (maybe) and returns that value?

Reply:
I introduced a dedicated methods on `Instance` and replaced other uses like that.